ClientLib/src/main/java/com/o3dr/services/android/lib/drone/companion/solo/tlv/TLVMessageParser.java

```diff
@@ -44,14 +44,20 @@
                 final int messageLength = packetBuffer.getInt();
                 Log.d(TAG, String.format("Received message %d of with value of length %d. Remaining buffer size is %d", messageType, messageLength, packetBuffer.remaining()));
 
+                if (messageLength > packetBuffer.remaining()) {
```

Comment:
@chaviw stores `packetBuffer.remaining()` into a variable since it's used more than once.